### PR TITLE
CASSANDRA-19793 Split out CassandraTypes into separate module

### PR DESCRIPTION
**Pull request by jberragan**

### CODE REVIEW

File: cassandra-analytics-core/src/main/java/org/apache/cassandra/bridge/CassandraBridgeFactory.java [hunk 1]
```diff
@@ -124,6 +124,12 @@ private static String bridgeResourceName(@NotNull String label)
         return "/bridges/" + label + "-bridge.jar";
     }
 
+    @NotNull
+    private static String typesResourceName(@NotNull String label)
+    {
+        return "/bridges/" + label + "-types.jar";
+    }
```
Comment (yifan-c) [inline cassandra-analytics-core/src/main/java/org/apache/cassandra/bridge/CassandraBridgeFactory.java]: nit: this method and the 2 methods above can be merged, e.g. 
```java
    private static String jarResourceName(String[] parts)
    {
        return "/bridges/" + String.join("-", parts) + ".jar";
    }

// usage
jarResourceName(label);
jarResourceName(label, "types);
```

Please feel free to ignore this nit.

File: cassandra-bridge/src/main/java/org/apache/cassandra/bridge/CassandraBridge.java [hunk 2]
```diff
@@ -200,65 +192,152 @@ public List<CqlField.NativeType> supportedTypes()
 
     // Native
 
-    public abstract CqlField.NativeType ascii();
+    public CqlField.NativeType ascii()
+    {
+        return cassandraTypes().ascii();
+    }
```
Comment (yifan-c) [inline cassandra-bridge/src/main/java/org/apache/cassandra/bridge/CassandraBridge.java:198]: How about only exposing `cassandraTypes()` from bridge and getting the specific type at the callsites? Maybe it comes in later refactoring?

File: cassandra-four-zero-types/build.gradle [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        junitXml {
+            enabled true
+            destination = destDir
+        }
+        html {
+            enabled true
+            destination = destDir
+        }
+    }
+}
```
Comment (yifan-c) [inline cassandra-four-zero-types/build.gradle]: add new line

File: cassandra-bridge/src/main/java/org/apache/cassandra/bridge/CassandraBridge.java (hunk 2 above)
Comment (jberragan) [reply cassandra-bridge/src/main/java/org/apache/cassandra/bridge/CassandraBridge.java:198]: I kept this as a convenience, but also to reduce the PR diff.

**Comment by jberragan** [comment]
CI green: https://app.circleci.com/pipelines/github/jberragan/cassandra-analytics/55/workflows/57c0c708-3d2f-418d-ab98-295aa68fa054

File: cassandra-four-zero-types/build.gradle [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    id('java-library')
+}
+
+configurations {
+    all*.exclude(group: 'org.slf4j', module: 'slf4j-log4j12')
+    all*.exclude(group: 'log4j', module: 'log4j')
+}
+
+dependencies {
+    compileOnly project(":cassandra-analytics-common")
```
Comment (frankgh) [inline cassandra-four-zero-types/build.gradle]: we don't have any dependencies in common for this project

File: cassandra-four-zero-types/build.gradle [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    compileOnly(project(':cassandra-bridge'))
+
+    compileOnly(group: "${sparkGroupId}", name: "spark-core_${scalaMajorVersion}", version: "${project.rootProject.sparkVersion}")
+    compileOnly(group: "${sparkGroupId}", name: "spark-sql_${scalaMajorVersion}", version: "${project.rootProject.sparkVersion}")
+
+    compileOnly(project(path: ':cassandra-four-zero', configuration: 'shadow'))
+    compileOnly "com.esotericsoftware:kryo-shaded:${kryoVersion}"
```
Comment (frankgh) [inline cassandra-four-zero-types/build.gradle]: I'm also not seeing any dependency on kryo for this subproject, can we remove it

File: cassandra-four-zero-types/build.gradle [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    testImplementation("org.junit.jupiter:junit-jupiter-api:${project.junitVersion}")
+    testImplementation("org.junit.jupiter:junit-jupiter-params:${project.junitVersion}")
+    testImplementation("org.junit.jupiter:junit-jupiter-engine:${project.junitVersion}")
```
Comment (frankgh) [inline cassandra-four-zero-types/build.gradle]: I am not seeing any tests in this subproject, should we have this here?

**Review by frankgh — COMMENTED** [review]
Looks good overall, but I have a couple of comments

File: cassandra-four-zero-types/build.gradle (hunk 6 above)
Comment (jberragan) [reply cassandra-four-zero-types/build.gradle]: Yes, it's a good point I can clean up these dependencies.

**Pull request merged**